waku/node/v2/wakunode2.nim

```diff
@@ -265,11 +263,24 @@ proc setFilterPeer*(n: WakuNode, address: string) =
 
   n.wakuFilter.setPeer(remotePeer)
 
-proc connectToNodes*(n: WakuNode, nodes: openArray[string]) =
+proc connectToNodes*(n: WakuNode, nodes: seq[string]) {.async.} =
   for nodeId in nodes:
     info "connectToNodes", node = nodeId
     # XXX: This seems...brittle
-    discard dialPeer(n, nodeId)
+    await dialPeer(n, nodeId)
+    
+  # This sleep is done to ensure we only subscribe when the protocol has been mounted.
+  # Meaning graft messages were received.
+  await sleepAsync(5.seconds)
```

Comment:
My guess would be we can lower this to 2s but we can also leave it as is for now

Reply:
I guess I don't understand the problem well enough, but it is strange that this sleep is required here (well, it shouldn't be required at all, but ok).
The comment doesn't make it any more clear to me, this part of the code is dialing peers. How is it related exactly to the subscribing?

Reply:
Ok I read this now:

> The real fix to the problem was sleeping after connecting to nodes, not after subscribing to the relay or anything like that. What I have done now is to do that in the connectToNodes function additionally I have stopped using switch.dial and used the wrapped function instead. I tested this end-to-end with the chat2 and it seems to work. I am happy with this code now if this explanation makes sense to you.

Still not clear to me what the originally problem is/was.
If we would like to merge this as is, can you update this sleep with a proper `TODO` comment explaining the problem and the current workaround by sleeping.